packages/rocketchat-api/server/v1/users.js

```diff
@@ -270,3 +270,66 @@ RocketChat.API.v1.addRoute('users.createToken', { authRequired: true }, {
 		return data ? RocketChat.API.v1.success({data}) : RocketChat.API.v1.unauthorized();
 	}
 });
+
+RocketChat.API.v1.addRoute('users.getPreferences', { authRequired: true }, {
+	get() {
+		const user = this.isUserFromParams() ? RocketChat.models.Users.findOneById(this.userId) : this.getUserFromParams();
+		if (user.settings) {
+			const preferences = user.settings.preferences;
+			preferences['language'] = user.language;
+
+			return RocketChat.API.v1.success({
+				preferences
+			});
+		} else {
+			return RocketChat.API.v1.failure(TAPi18n.__('Accounts_Default_User_Preferences').toUpperCase());
+		}
+	}
+});
+
+RocketChat.API.v1.addRoute('users.setPreferences', { authRequired: true }, {
+	post() {
+		check(this.bodyParams, {
+			userId: String,
```

Comment:
Why need to pass the userId? Would be able to save preferences for other users?

Reply:
you can set preferences only to your own user or if you are an admin to other users, this is handled on `packages/rocketchat-lib/server/functions/saveUser.js:75`

Reply:
would it be better to maybe not require it, and default to the user registered if none is passed?

Reply:
good catch @geekgonecrazy !!
performance wise it doesn't change much, but logic wise it makes more sense. 
committing that right away.